src/main/java/org/wso2/carbon/apimgt/securityenforcer/executors/PingAIExecutor.java

```diff
@@ -210,7 +220,17 @@ private GenericArtifactManager getArtifactManager(Registry registry, String key)
     }
 
     private JSONObject createAPIJSON(String APIContext, RequestContext requestContext) {
-        JSONObject managmentAPIPayload = ServiceReferenceHolder.getInstance().getManagementAPIPayload();
+        JSONParser jsonParser = new JSONParser();
```

Comment:
We have observed, when the first API is published with additional properties, those properties are being updated and retained in the **managmentAPIPayload**. Even if the next API's don't have any additional properties, the properties of previously published API's are being applied to the subsequent ones. This is the reason for loading the json file each time.

Reply:
As most of the time, API publisher will use the default configurations I think it unnecessary to do a file read every time when an API is published (As we cannot change any fields in the JSON, always we will read the same file again and again). Rather than using this approach, we can read the JSON file once and record all the default configuration to a class. Whenever we need to send the details, create an object of this class and if there are any fields to be updated, we can update it. We can parse this object to JSON easily.

Reply:
Created a deep copy of managmentAPIPayload instead of loading sampleAPIJson everytime.